core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -63,7 +63,8 @@ protected BaseMetadataTable(TableOperations ops, Table table, String name) {
    */
   static PartitionSpec transformSpec(Schema metadataTableSchema, PartitionSpec spec, String partitionPrefix) {
     PartitionSpec.Builder identitySpecBuilder = PartitionSpec.builderFor(metadataTableSchema);
```

Comment:
call checkConflict here instead of in the loop, it will also make it all fit on one line below :)

Reply:
Fixed. Thanks!